widget/embedded/src/utils/event.ts

```diff
@@ -0,0 +1,134 @@
+import type { SelectedQuote, Wallet } from '../types';
+import type { WalletType } from '@rango-dev/wallets-shared';
+import type { BlockchainMeta, Token } from 'rango-sdk';
+
+import {
+  QuoteEventTypes,
+  WalletEventTypes,
+  WidgetEvents,
+} from '../types/event';
+
+export function getQuoteSpecificEventEmitter(
+  eventEmitter: any,
+  eventType: QuoteEventTypes.FROM_BLOCKCHAIN_CHANGED
+): (blockchain: BlockchainMeta | null) => void;
+export function getQuoteSpecificEventEmitter(
+  eventEmitter: any,
+  eventType: QuoteEventTypes.FROM_TOKEN_CHANGED
+): (token: Token | null) => void;
+export function getQuoteSpecificEventEmitter(
+  eventEmitter: any,
+  eventType: QuoteEventTypes.TO_BLOCKCHAIN_CHANGED
+): (blockchain: BlockchainMeta | null) => void;
+export function getQuoteSpecificEventEmitter(
+  eventEmitter: any,
+  eventType: QuoteEventTypes.TO_TOKEN_CHANGED
+): (token: Token | null) => void;
+export function getQuoteSpecificEventEmitter(
+  eventEmitter: any,
+  eventType: QuoteEventTypes.INPUT_AMOUNT_CHANGED
+): (amount: string | null) => void;
+export function getQuoteSpecificEventEmitter(
+  eventEmitter: any,
+  eventType: QuoteEventTypes.SELECTED_QUOTE_CHANGED
+): (quote: SelectedQuote | null) => void;
+export function getQuoteSpecificEventEmitter(
+  eventEmitter: any,
+  eventType: QuoteEventTypes
+) {
```

Comment:
Using any removes all the benefits of TypeScript.
Also I think declaring all these function overloads and helpers is excessive and makes things more complex. I don't see any problem in using `EventEmitter` without any helpers.